src/renderer/modals/loadingScreen/index.tsx

```diff
@@ -1,65 +1,60 @@
 // Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react';
+import React, {useState, useEffect} from 'react';
 import ReactDOM from 'react-dom';
 
+import {useConfig} from 'renderer/hooks/useConfig';
+
 import LoadingScreen from '../../components/LoadingScreen';
 
 import 'renderer/css/components/LoadingAnimation.css';
 import 'renderer/css/components/LoadingScreen.css';
 
-type Props = Record<string, never>;
-
-type State = {
-    showLoadingScreen: boolean;
-    darkMode: boolean;
-}
+const LoadingScreenRoot: React.FC = () => {
+    const [showLoadingScreen, setShowLoadingScreen] = useState(true);
+    const {config} = useConfig();
 
-class LoadingScreenRoot extends React.PureComponent<Props, State> {
-    constructor(props: Props) {
-        super(props);
-        this.state = {
-            showLoadingScreen: true,
-            darkMode: false,
+    useEffect(() => {
+        const handleToggleLoadingScreenVisibility = (showLoadingScreen: boolean) => {
+            setShowLoadingScreen(showLoadingScreen);
         };
-    }
-
-    async componentDidMount() {
-        window.desktop.onDarkModeChange(this.setDarkMode);
-        const darkMode = await window.desktop.getDarkMode();
-        this.setDarkMode(darkMode);
 
-        window.desktop.loadingScreen.onToggleLoadingScreenVisibility(this.onToggleLoadingScreenVisibility);
+        const initializeApp = async () => {
+            window.desktop.loadingScreen.onToggleLoadingScreenVisibility(handleToggleLoadingScreenVisibility);
 
-        window.addEventListener('click', () => {
-            window.desktop.closeServersDropdown();
-            window.desktop.closeDownloadsDropdown();
-        });
-    }
+            window.addEventListener('click', () => {
+                window.desktop.closeServersDropdown();
+                window.desktop.closeDownloadsDropdown();
+            });
+        };
 
-    setDarkMode = (darkMode: boolean) => {
-        this.setState({darkMode});
-    };
+        initializeApp();
 
-    onToggleLoadingScreenVisibility = (showLoadingScreen: boolean) => {
-        this.setState({showLoadingScreen});
-    };
+        // Cleanup function
+        return () => {
+            // Note: In a real app, you might want to remove event listeners here
+            // but since this is a root component that doesn't unmount, it's not strictly necessary
+        };
+    }, []);
 
-    onFadeOutComplete = () => {
+    const onFadeOutComplete = () => {
```

Comment:
Same comment about passing this function later in a prop. Either move it out of the component, or add a useCallback.